docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,926 @@
+		return p.FeeManagerConfig, p.FeeManagerConfig != nil
+	case helloWorldKey:
+		return p.HelloWorldConfig, p.HelloWorldConfig != nil
+		// ADD YOUR PRECOMPILE HERE
+	/*
+		case {yourPrecompile}Key:
+		return p.{YourPrecompile}Config , p.{YourPrecompile}Config  != nil
+	*/
+	default:
+		panic(fmt.Sprintf("unknown upgrade key: %v", key))
+	}
+}
+```
+
+We should also define `GetHelloWorldConfig`. Given a `blockTimestamp`, we will return the `*precompile.HelloWorldConfig` if it is enabled. We use this function to check whether precompiles are enabled in `params/config.go`. This function is also used to construct a `PrecompileUpgrade` struct in `GetActivePrecompiles()`.
+
+```go
+// GetHelloWorldConfig returns the latest forked HelloWorldConfig
+// specified by [c] or nil if it was never enabled.
+func (c *ChainConfig) GetHelloWorldConfig(blockTimestamp *big.Int) *precompile.HelloWorldConfig {
+	if val := c.getActivePrecompileConfig(blockTimestamp, helloWorldKey, c.PrecompileUpgrades); val != nil {
+		return val.(*precompile.HelloWorldConfig)
+	}
+	return nil
+}
+```
+
+Finally, we can add our precompile in `GetActivePrecompiles()`, which given a `blockTimestamp` returns a `PrecompileUpgrade` struct. This function is used in the Ethereum API and can give a user information about what precompiles are enabled at a certain block timestamp. For more information about our Ethereum API, check out this [link](https://docs.avax.network/apis/avalanchego/apis/subnet-evm).
+
+```go
+func (c *ChainConfig) GetActivePrecompiles(blockTimestamp *big.Int) PrecompileUpgrade {
+	pu := PrecompileUpgrade{}
+	if config := c.GetContractDeployerAllowListConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.ContractDeployerAllowListConfig = config
+	}
+	if config := c.GetContractNativeMinterConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.ContractNativeMinterConfig = config
+	}
+	if config := c.GetTxAllowListConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.TxAllowListConfig = config
+	}
+	if config := c.GetFeeConfigManagerConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.FeeManagerConfig = config
+	}
+  if config := c.GetHelloWorldConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.HelloWorldConfig = config
+	}
+	// ADD YOUR PRECOMPILE HERE
+	// if config := c.{YourPrecompile}Config(blockTimestamp); config != nil && !config.Disable {
+	// 	pu.{YourPrecompile}Config = config
+	// }
+
+	return pu
+}
+```
+
+Done! All we had to do was follow the comments.
+
+### Step 5: Add Precompile Upgrade
+
+Let's add our precompile upgrade in `params/config.go`. We can `CTRL F` for `ADD YOUR PRECOMPILE HERE`. This file is used to set up blockchain settings.
+
+Let's add the bool to check if our precompile is enabled. We are adding this to the `Rules` struct. `Rules` gives information about the blockchain, the version, and the precompile enablement status to functions that don't have this information.
```

Comment:
The order of code change is different from that in the code.

Reply:
Unfortunately I had to change the ordering because we need to write the function `IsHelloWorld` before we set and define  it as a rule.